pyramid/scripts/pserve.py

```diff
@@ -887,11 +889,28 @@ def check_reload(self):
                 continue
             if filename.endswith('.pyc') and os.path.exists(filename[:-1]):
                 mtime = max(os.stat(filename[:-1]).st_mtime, mtime)
+                pyc = True
+            else:
+                pyc = False
             if filename not in self.module_mtimes:
                 self.module_mtimes[filename] = mtime
             elif self.module_mtimes[filename] < mtime:
-                print("%s changed; reloading..." % filename)
-                return False
+                if pyc:
+                    filename = filename[:-1]
+                # check if a file has syntax errors. If so, track it until it's fixed
+                for filename_promise in self.syntax_error_files + [filename]:
+                    try:
+                        py_compile.compile(filename_promise, doraise=True)
+                    except py_compile.PyCompileError:
+                        print("%s has a SyntaxError; NOT reloading." % filename_promise)
```

Comment:
> Why is logging wrong? Logging is user-configurable and allows you to agnostically write to the user's preferred output. As far as informational versus erroneous conditions, that why log levels exist -- you can use info() and error().

The logging is for the server itself, not for the runner. This is information the runner is providing, and is not at all related to the application/WSGI server itself.

I think that logging that with the logger is wrong, I don't expect output from my runner to show up in my logger/logging.

This is also related to a very small part of the runner, the `--reload` feature, which IMHO should not be used in anything but development.